0.preprocess-sites/4.image-and-segmentation-qc.py

```diff
@@ -317,32 +317,59 @@
 saturated_col_prefix = "ImageQuality_PercentMaximal_"
 cp_sat_cols = []
 bc_sat_cols = []
+addn_cols = ["Metadata_Site_Full", "Fails_CP_Sat", "Fails_BC_Sat"]
 nts = ["A", "C", "G", "T"]
+cp_sat_df = pd.DataFrame()
+bc_sat_df = pd.DataFrame()
+sat_df = pd.DataFrame()
 
+# Create list of columns measuring CP saturation
 for name in painting_image_names:
     cp_sat_cols.append(f"{saturated_col_prefix}{name}")
 
+# Create list of columns measuring BC saturation
 for x in range(1, (barcoding_cycles + 1)):
     for nt in nts:
         bc_sat_cols.append(f"{saturated_col_prefix}{barcoding_prefix}{x:02d}_{nt}")
 
+# Create df of sites that fail CP saturation
 if all(x in image_df.columns.tolist() for x in cp_sat_cols):
     for col in cp_sat_cols:
-        cp_sat_df = image_df[image_df[col] > 1]
+        temp = image_df[image_df[col] > 1]
+        cp_sat_df = cp_sat_df.append(temp)
+    if len(cp_sat_df) != 0:
+        cp_sat_df.loc[:, "Fails_CP_Sat"] = "Fails"
 
-    for col in cp_sat_cols:
-        cp_sat_df = image_df[image_df[col] > 1]
+# Create df of sites that fail BC saturation
+if all(x in image_df.columns.tolist() for x in bc_sat_cols):
     for col in bc_sat_cols:
-        bc_sat_df = image_df[image_df[col] > 0.25]
-    sat_df_cols = cp_sat_cols + bc_sat_cols
-    sat_df_cols.append("site")
-
-    sat_df = cp_sat_df.append(bc_sat_df).drop_duplicates(subset="site")
-
-    if len(sat_df.index) > 0:
-        sat_output_file = pathlib.Path(results_output, "saturated_sites.csv")
-        if check_if_write(output_file, force, throw_warning=True):
-            sat_df.to_csv(sat_output_file)
+        temp = image_df[image_df[col] > 0.25]
+        bc_sat_df = bc_sat_df.append(temp)
+    if len(bc_sat_df) != 0:
+        bc_sat_df.loc[:, "Fails_BC_Sat"] = "Fails"
+
+if len(cp_sat_df) > 0 and len(bc_sat_df) > 0:
+    sat_df = (
+        cp_sat_df.set_index("Metadata_Site_Full")
```

Comment:
is `Metadata_Site_Full` defined somewhere else as an important column? What would happen if this column isn't present?

Reply:
See comment above for where `Metadata_Site_Full` is hardcoded. Therefore okay to hardcode here?

Reply:
i think so, but we should try to be internally consistent with nomenclature moving forward. This is my bad! We should have thought through this decision in more detail together - would have saved some time